### DISCUSSION TITLE
On submitting an invalid form the form resets to default value

### DISCUSSION THREAD

**Discussion by cyrilgeorgemv** [react-hook-form/react-hook-form#4961] [Q&A]
Can some one give me pointers on possible scenarios where the form resets to the default value when there is a validation failure .

I am unable to reproduce the same in codesandbox ,so it must be the way I am using the lib.
 https://codesandbox.io/s/epic-bush-p6hwd?file=/src/App.js

Sorry if the question too vague, can't really share more info on the form or code due to workplace restrictions.

Using react-hook-form v6.15.5

**Comment by bluebill1049** [comment]
when there is a validation failure? what do you mean during `onChange` or after submit?

**Reply by cyrilgeorgemv** [reply]
After submit. When invoking handleSumbit

**Reply by bluebill1049** [reply]
Is this what you after?

```ts
useEffect(() => {
 if (formState.isSubmitted && Object.keys(formState.errors).length) {
   reset({
     test: 'test'
   })
 }
}, [reset, formState])
```

**Reply by cyrilgeorgemv** [reply]
On handlesubmit when there are validation errors,the form resets to default value, all the user changes gets cleared . This is the current behavior I am getting. I don't want that, for some reason it's resetting
